nowcasting_dataset/manager.py

```diff
@@ -0,0 +1,384 @@
+        `config.process.upload_every_n_batches > 0`.
+      local_temp_path: Path: `config.process.local_temp_path` with `~` expanded.
+    """
+
+    def __init__(self) -> None:  # noqa: D107
+        self.config = None
+        self.data_sources = {}
+        self.data_source_which_defines_geospatial_locations = None
+
+    def load_yaml_configuration(self, filename: str) -> None:
+        """Load YAML config from `filename`."""
+        logger.debug(f"Loading YAML configuration file {filename}")
+        self.config = config.load_yaml_configuration(filename)
+        self.config = config.set_git_commit(self.config)
+        self.save_batches_locally_and_upload = self.config.process.upload_every_n_batches > 0
+
+        # TODO: This should probably be done in the Pydantic model.
+        self.local_temp_path = Path(self.config.process.local_temp_path).expanduser()
+        logger.debug(f"config={self.config}")
+
+    def initialise_data_sources(
+        self, names_of_selected_data_sources: Optional[list[str]] = ALL_DATA_SOURCE_NAMES
+    ) -> None:
+        """Initialise DataSources specified in the InputData configuration.
+
+        For each key in each DataSource's configuration object, the string `<data_source_name>_`
+        is removed from the key before passing to the DataSource constructor.  This allows us to
+        have verbose field names in the configuration YAML files, whilst also using standard
+        constructor arguments for DataSources.
+        """
+        for data_source_name in names_of_selected_data_sources:
+            logger.debug(f"Creating {data_source_name} DataSource object.")
+            config_for_data_source = getattr(self.config.input_data, data_source_name)
+            if config_for_data_source is None:
+                logger.info(f"No configuration found for {data_source_name}.")
+                continue
+            config_for_data_source = config_for_data_source.dict()
+
+            # Strip `<data_source_name>_` from the config option field names.
+            config_for_data_source = nd_utils.remove_regex_pattern_from_keys(
+                config_for_data_source, pattern_to_remove=f"^{data_source_name}_"
+            )
+
+            data_source_class = MAP_DATA_SOURCE_NAME_TO_CLASS[data_source_name]
+            try:
+                data_source = data_source_class(**config_for_data_source)
+            except Exception:
+                logger.exception(f"Exception whilst instantiating {data_source_name}!")
+                raise
+            self.data_sources[data_source_name] = data_source
+
+        # Set data_source_which_defines_geospatial_locations:
+        try:
+            self.data_source_which_defines_geospatial_locations = self.data_sources[
+                self.config.input_data.data_source_which_defines_geospatial_locations
+            ]
+        except KeyError:
+            logger.error(
```

Comment:
should this raise an error?

Reply:
Good catch!  (I had thought that `logger.error` raised an exception but I've just tried and it doesn't!)  Fixed.